client/python/gradio_client/serializing.py

```diff
@@ -291,6 +300,24 @@ def _deserialize_single(
                     )
                 else:
                     file_name = utils.create_tmp_copy_of_file(filepath, dir=save_dir)
+            elif x.get("is_stream"):
+                assert x["name"] and root_url and save_dir
+                if not self.stream:
+                    self.stream = self._setup_stream(
+                        root_url + "stream/" + x["name"], hf_token=hf_token
+                    )
+                try:
+                    chunk = next(self.stream)
+                except StopIteration:
+                    self.stream = self._setup_stream(
+                        root_url + "stream/" + x["name"], hf_token=hf_token
+                    )
+                    chunk = next(self.stream)
```

Comment:
Sorry can you explain why these two lines are repeated from above?

Reply:
It was to determine when the stream changed but there's a better/clearer way. Will push up now!